schema/resources.json

```diff
@@ -0,0 +1,100 @@
+{
+    "$id": "https://serverlessworkflow.io/schemas/0.8/resources.json",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "description": "Serverless Workflow specification - resources schema",
+    "type": "object",
+    "resources":{
+        "type": "array",
+        "description": "Defines the external resources referenced by the workflow",
+        "items": {
+          "type": "object",
+          "$ref": "#/definitions/resourceRef"
+        },
+        "additionalItems": false,
+        "minItems": 1
+      },
+    "required":[
+        "resources"
+    ],
+    "definitions":{
+        "resourceRef":{
+            "type": "object",
+            "description": "References an external resource",
+            "properties": {
+              "name": {
+                "type": "string",
+                "description": "The name of the external resource. It must be lowercased and must only contain alphanumeric characters, with the exception of `-` and `_`. The name is used as prefix to reference any definition contained in the referenced resource.",
+                "pattern": "^[a-z0-9-_]+$"
```

Comment:
Can we please remove this pattern for now, and then apply a pattern across all spec props used for lookups.

Reply:
This PR is now pending merge of #768, so I'll fix resources to match the pattern specified there